### DISCUSSION TITLE
[Bug] banyan standalone mode need connect to etcd

### DISCUSSION THREAD

**Discussion by mindfocus** [apache/skywalking#12554] [Q&A] [answered]
banyandb 0.6.1

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

BanyanDB (apache/skywalking-banyandb)

### What happened

{"level":"info","time":"2024-08-22T14:25:06+08:00","message":"starting as a standalone server"}
{"level":"info","module":"STANDALONE","time":"2024-08-22T14:25:06+08:00","message":"started"}
{"level":"info","module":"STANDALONE","roles":[1,2,3],"time":"2024-08-22T14:25:06+08:00","message":"the node will run as"}
{"level":"info","module":"STANDALONE","elapsed":0.15447,"name":"signal","time":"2024-08-22T14:25:06+08:00","message":"pre-run completed"}
{"level":"warn","module":"METADATA","etcd-endpoints":["http://localhost:2379"],"time":"2024-08-22T14:25:18+08:00","message":"register node timeout, retrying..."}
{"level":"warn","module":"METADATA","etcd-endpoints":["http://localhost:2379"],"time":"2024-08-22T14:25:28+08:00","message":"register node timeout, retrying..."}

### What you expected to happen

do not connect to ectd

### How to reproduce

I don't know either. It was running fine half a month ago, until one day the OAP service filled up the hard disk with error logs, and that's when I realized that it could no longer connect to BanyanDB. That's when I traced the issue to this error. When I tried to restart BanyanDB using ./banyand-server-static standalone, it started trying to connect to etcd.

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by mindfocus** [comment]
[nohup.zip](https://github.com/user-attachments/files/16706360/nohup.zip)

log here

**Comment by hanahmily** [comment, marked as answer]
Based on your logs, the standalone server uses an embed "etc" that is not booting up properly. If you encounter a disk-full issue, the `metadata` files may crash. In this case, you can manually delete this data. The location is specified by the flag "--metadata-root-path," which defaults to "/tmp/metadata."

The OAP will create these metadata in its booting procedure

**Reply by mindfocus** [reply]
Thank you for your response. The issue has been resolved. I'm a bit confused. It seems that BanyanDB crashed first, and then the OAP service kept retrying the connection, which filled up the disk with retry logs. So the sequence of events is actually the opposite of what you mentioned. However, the conclusion should be the same—the metadata file crashed.

**Comment by mindfocus** [comment]
I'm a bit confused. It seems that BanyanDB crashed first, and then the OAP service kept retrying the connection, which filled up the disk with retry logs. So the sequence of events is actually the opposite of what you mentioned. However, the conclusion should be the same—the metadata file crashed.

**Reply by wu-sheng** [reply]
If you shared OAP logs and BanyanDB data files in the same place. This could happen.

**Reply by mindfocus** [reply]
Thank you for your response. The issue has been resolved, but I still don't understand why banyandb crashed first. Additionally, another concern is that the log output frequency of oap seems to be quite high.

**Reply by wu-sheng** [reply]
That is the case. Of course OAP is going to retry.
